pkg/storer/internal/reserve/reserve.go

```diff
@@ -100,9 +101,12 @@ func New(
 //     if the new chunk has a higher stamp timestamp (regardless of batch type and chunk type, eg CAC & SOC).
 func (r *Reserve) Put(ctx context.Context, chunk swarm.Chunk) error {
 
+	chunkType := storage.ChunkType(chunk)
+
 	// batchID lock, Put vs Eviction
-	r.multx.Lock(string(chunk.Stamp().BatchID()))
```

Comment:
Why do you need to lock by stamp?

Reply:
original idea was that if somehow a chunk with mutable batch stamped twice that have the same SOC address then it will do not let to put it into the reserve in the same time.
anyway you are right, I reverted this change since the batchID locking is higher level locking.